rethinkdb/datadog_checks/rethinkdb/check.py

```diff
@@ -32,21 +32,48 @@ def __init__(self, *args, **kwargs):
         if self.config.password:
             self.register_secret(self.config.password)
 
-        self.queries = (
-            queries.config_summary,
-            queries.cluster_statistics,
-            queries.server_statistics,
-            queries.table_statistics,
-            queries.replica_statistics,
-            queries.table_statuses,
-            queries.server_statuses,
-            queries.jobs_summary,
-            queries.current_issues_summary,
-        )  # type: Sequence[DocumentQuery]
+        self._conn = None  # type: Optional[rethinkdb.net.Connection]
+
+        manager_queries = [
+            queries.ClusterMetrics,
+            queries.ServerMetrics,
+            queries.DatabaseMetrics,
+            queries.DatabaseTableMetrics,
+            queries.TableMetrics,
+            queries.ReplicaMetrics,
+            queries.ShardMetrics,
+            queries.JobMetrics,
+            queries.CurrentIssuesMetrics,
+        ]  # type: list
+
+        if self.is_metadata_collection_enabled:
+            manager_queries.append(queries.VersionMetadata)
+
+        self._query_manager = QueryManager(
+            self,
+            executor=self._execute_raw_query,
+            queries=manager_queries,
+            tags=self.config.tags,
+        )
+        self._query_funcs = {}  # type: Dict[str, Callable]
+
+        self.check_initializations.append(self._query_manager.compile_queries)
+
+    def _execute_raw_query(self, query):
+        # type: (str) -> List[tuple]
+        # TODO: change this once datadog_checks_base.utils.db supports `'query': <callable>`
```

Comment:
Actually not sure that we need to add this to `datadog_checks_base` at all, although it would be "prettier" (and easier to navigate from IDEs) to be able to just import query functions in the `queries.py` module directly.